pkg/builds/pipeline.go

```diff
@@ -105,12 +106,16 @@ func RunPipeline(ctx context.Context, pr *tknv1beta1.PipelineRun, opt *options.L
 // which configures a temporary ServiceAccount infused with the local credentials
 // for the container registry hosting the image we will publish to (and to which
 // the source is published).
-func WithPipelineServiceAccount(sa string, refs ...name.Reference) CancelablePipelineOption {
-	cfg, err := GetConfig("", "")
-	if err != nil {
-		log.Fatal("GetConfig() =", err)
+func WithPipelineServiceAccount(ctx context.Context, sa string, refs ...name.Reference) CancelablePipelineOption {
+	client := kubeclient.Get(ctx)
+	// TODO replace once injection is sorted?
+	if client == nil {
```

Comment:
Let me try to clean up the context injection stuff tonight, since I have a picture in my head, and I think it'll cut down on what you need here (tests + namespace stuff) significantly if I can land that 🤞

Reply:
Alright since there's an in-flight PR to update cobra in tekton and drop the fork, I staged things in a way that we can hopefully pick up ExecuteContext easily here: https://github.com/mattmoor/mink/pull/383.

I'll try to land that tonight, but there's another change in front of it to fix eventing, which had a breaking change.